### ISSUE TITLE
SSH Agent does not add keys on first unlock if --pw-stdin is used

### ISSUE THREAD

**Issue by hifi** [keepassxreboot/keepassxc#2912]
Based on comment by @DevOliv in https://github.com/keepassxreboot/keepassxc/issues/2902#issuecomment-478215316

## Expected Behavior
Unlock adds automatic keys with --pw-stdin.

## Current Behavior
Nothing happens.

## Steps to Reproduce
1. Have keys with auto-add on unlock
2. `echo 'test' | keepassxc test.kdbx`
3. No added keys, unlock & lock adds them.

## Context
Regression from 2.3.

## Debug Info
KeePassXC - Version 2.4.0-snapshot
Build Type: Snapshot
Revision: c51752d

Libraries:
- Qt 5.11.1
- libgcrypt 1.8.3

Operating system: Ubuntu 18.10
CPU architecture: x86_64
Kernel: linux 4.18.0-16-generic

Enabled extensions:
- Auto-Type
- SSH Agent
- KeeShare (only unsigned sharing)

**Comment by hifi** [comment]
```
d612cad09a4f1cd8ef9bf6be8bf90b75248ecfef is the first bad commit
commit d612cad09a4f1cd8ef9bf6be8bf90b75248ecfef
Author: Janek Bevendorff <janek@jbev.net>
Date:   Thu Nov 22 11:47:31 2018 +0100

    Refactor Database and Database widgets (#2491)
    
    The Database, DatabaseWidget, and DatabaseTabWidget classes share many responsibilities in inconsistent ways resulting in impenetrable and unmaintainable code and a diverse set of bugs and architecture restrictions. This patch reworks the architecture, responsibilities of, and dependencies between these classes.
    
    The core changes are:
    
    * Move loading and saving logic from widgets into the Database class
    * Get rid of the DatabaseManagerStruct and move all the information contained in it into the Database
    * Let database objects keep track of modifications and dirty/clean state instead of handing this to external widgets
    * Move GUI interactions for loading and saving from the DatabaseTabWidget into the DatabaseWidget (resolves #2494 as a side-effect)
    * Heavily clean up DatabaseTabWidget and degrade it to a slightly glorified QTabWidget
    * Use QSharedPointers for all Database objects
    * Remove the modifiedImmediate signal and replace it with a markAsModified() method
    * Implement proper tabName() method instead of reading back titles from GUI widgets (resolves #1389 and its duplicates #2146 #855)
    * Fix unwanted AES-KDF downgrade if database uses Argon2 and has CustomData
    * Improve code
    
    This patch is also the first major step towards solving issues #476 and #2322.

:040000 040000 9b757359d3f0fd2e8aae2ac712e6f49b3a5ebcfc 07f86848fa8d6727859128d10342ab840f6a01e4 M      src
:040000 040000 5165a78bb093ca3785df9447431958f1d2b8562d 1bdc1bc1f351c0e6366d07163e39f65ee4c46da0 M      tests
```

Not surprising since the agent hasn't seen big refactors itself between 2.3 and 2.4.

**Comment by hifi** [comment]
28a3824d2d594429b38ca441ff0a718b6ec7092a is the first bad commit
commit 28a3824d2d594429b38ca441ff0a718b6ec7092a
```
Author: Jonathan White <support@dmapps.us>
Date:   Thu Mar 21 17:36:57 2019 -0400

    Use existing database open function on startup
    
    * Fix #2828

:040000 040000 140a692138a0637d199a3f6978495a469e30138e 8ff4ee191af239f9dce65f1ed9bfab8f54ac040b M      src
```

This was already fixed as part of the release/2.4.1 branch so the fix is already in. Closing.
